mopidy/files/library.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import unicode_literals
+
+import logging
+import operator
+import os
+import stat
+import sys
+import urllib2
+
+
+from mopidy import backend, exceptions, models
+from mopidy.audio import scan, utils
+from mopidy.internal import path
+
+logger = logging.getLogger(__name__)
+
+
+class FilesLibraryProvider(backend.LibraryProvider):
+    """Library for browsing local files."""
+
+    @property
+    def root_directory(self):
+        if not self._media_dirs:
+            return None
+        elif len(self._media_dirs) == 1:
+            localpath = self._media_dirs[0]['path']
+            uri = path.path_to_uri(localpath)
+        else:
+            uri = u'file:root'
+        return models.Ref.directory(name='Files', uri=uri)
+
+    def __init__(self, backend, config):
+        super(FilesLibraryProvider, self).__init__(backend)
+        self._media_dirs = []
+        # import pdb; pdb.set_trace()
+        for entry in config['files']['media_dir']:
+            media_dir = {}
+            media_dict = entry.split(':')
+            local_path = path.expand_path(
+                media_dict[0].encode(sys.getfilesystemencoding()))
+            st = os.stat(local_path)
+            if not stat.S_ISDIR(st.st_mode):
+                logger.warn(u'%s is not a directory' % local_path)
+                continue
+            media_dir['path'] = local_path
+            if len(media_dict) == 2:
+                media_dir['name'] = media_dict[1]
+            else:
+                media_dir['name'] = media_dict[0].replace(os.sep, '+')
+            self._media_dirs.append(media_dir)
+        logger.debug(self._media_dirs)
+        self._follow_symlinks = config['files']['follow_symlinks']
+        self._show_hidden = config['files']['show_hidden']
+        self._scanner = scan.Scanner(
+            timeout=config['files']['metadata_timeout'])
+
+    def browse(self, uri, encoding=sys.getfilesystemencoding()):
```

Comment:
Why the extra encoding param? I don't think we will ever call this with anything more than the URI.